src/h5web/dataset-visualizer/utils.ts

```diff
@@ -46,3 +47,20 @@ export function getSupportedVis(dataset?: HDF5Dataset): Vis[] {
   // Remove Raw vis if any other vis is supported
   return supported.length > 1 ? supported.slice(1) : supported;
 }
+
+export function useActiveVis(
+  supportedVis: Vis[]
+): [Vis | undefined, (vis: Vis) => void] {
+  const [activeVis, setActiveVis] = useState<Vis>();
+
+  // When switching between two datasets, `activeVis` may become stale for one render
+  const isStale = !activeVis || !supportedVis.includes(activeVis);
+
+  useEffect(() => {
+    if (isStale) {
+      setActiveVis(supportedVis[supportedVis.length - 1]);
+    }
+  }, [isStale, supportedVis]);
+
+  return [isStale ? undefined : activeVis, setActiveVis];
```

Comment:
`activeVis` is now consistently `undefined` both on first render and on stale renders.